lesson4/hometask4_mkolesnik.rb

```diff
@@ -0,0 +1,42 @@
+class LazyHash < Hash
+
+  def method_missing(method_name, *args, &block)
+    if !self[*args] && self.keys.include?(method_name) && lazy?(method_name)
+      if block_given?
```

Comment:
Could you explain what happens in every context of the condition? It looks really over-engineered and I don't sure that it works properly. Need more information for proper review.